frontend/src/main/scala/bloop/engine/tasks/ScalaNative.scala

```diff
@@ -0,0 +1,88 @@
+package bloop.engine.tasks
+
+import java.net.URLClassLoader
+import java.nio.file.{Files, Path}
+
+import bloop.{DependencyResolution, Project}
+import bloop.cli.ExitStatus
+import bloop.engine.State
+import bloop.exec.Process
+import bloop.io.AbsolutePath
+import bloop.logging.Logger
+
+import monix.eval.Task
+
+object ScalaNative {
+
+  /**
+   * Compile down to native binary using Scala Native's toolchain.
+   *
+   * @param project The project to link
+   * @param entry   The fully qualified main class name
+   * @param logger  The logger to use
+   * @return The absolute path to the native binary.
+   */
+  def nativeLink(project: Project, entry: String, logger: Logger): AbsolutePath = {
+
+    initializeToolChain(logger)
+
+    val bridgeClazz = nativeClassLoader.loadClass("bloop.scalanative.NativeBridge")
+    val paramTypes = classOf[Project] :: classOf[String] :: classOf[Logger] :: Nil
+    val nativeLinkMeth = bridgeClazz.getMethod("nativeLink", paramTypes: _*)
+
+    // The Scala Native toolchain expects to receive the module class' name
+    val fullEntry = if (entry.endsWith("$")) entry else entry + "$"
+
+    AbsolutePath(nativeLinkMeth.invoke(null, project, fullEntry, logger).asInstanceOf[Path])
```

Comment:
Can we protect us here from a potential null returned by `invoke` and only instantiate an absolute path if we're 100% sure the reflective execution didn't fail?

Reply:
I think it is cleaner to pass the target path to `nativeLink()`. Then, we can ignore the result altogether.